apps/src/netsim/NetSimMessage.js

```diff
@@ -137,28 +181,10 @@ NetSimMessage.prototype.getTable = function () {
 };
 
 /**
- * @typedef {Object} MessageRow
- * @property {number} fromNodeID - this message in-flight-from node
- * @property {number} toNodeID - this message in-flight-to node
- * @property {number} simulatedBy - Node ID of the client responsible for
- *           all operations involving this message.
- * @property {Base64Payload} base64Payload - base64-encoded binary
- *           message content, all of which can be exposed to the
- *           student.  May contain headers of its own.
- */
-
-/**
  * Build own row for the message table
  * @returns {MessageRow}
  * @throws {TypeError} if payload is invalid
  */
 NetSimMessage.prototype.buildRow = function () {
-  return {
-    fromNodeID: this.fromNodeID,
-    toNodeID: this.toNodeID,
-    simulatedBy: this.simulatedBy,
-    base64Payload: binaryToBase64(this.payload),
-    extraHopsRemaining: this.extraHopsRemaining,
-    visitedNodeIDs: this.visitedNodeIDs
-  };
+  return NetSimMessage.buildRowFromData(this);
```

Comment:
I ask because you're passing `this` (which is `NetSimMessage`) to a method that expects `MessageData`, requiring that the one implements the other.  I think the `NetSimMessage` tests already have some tests that ensure certain members exist - might want to double-check that they cover all `MessageData` properties.